CMakeLists.txt

```diff
@@ -757,9 +757,12 @@ target_link_libraries (seastar
     lksctp-tools::lksctp-tools
     rt::rt
     yaml-cpp::yaml-cpp
-    Valgrind::valgrind
     Threads::Threads)
 
+target_include_directories (seastar
+  PRIVATE
+    ${Valgrind_INCLUDE_DIRS})
```

Comment:
i see. so i'd suggest make following changes

1. change the title of commit message to something like: "build: do not expose Valgrind in SeastarTargets.cmake"
1. in the commit message
   - explain the problem you ran into. and explain the root cause: despite that the interface library is claimed PRIVATE, it is still gets populated to `SeastarTargets.cmake`. 
   - reference https://gitlab.kitware.com/cmake/cmake/-/issues/15415
1. in `CMakeLists.txt`, change `Valgrind::valgrind` to `"$<BUILD_INTERFACE:Valgrind::valgrind>"`. so `Valgrind::valgrind` is only exposed when `seastar` is being built or when it is being linked by another target in the same build system.

Reply:
I updated commit. I hope I got your idea right. Using `"$<BUILD_INTERFACE:Valgrind::valgrind>"` removes transitive dependency from installation cmake finders.